MLP(
  (layer1): Sequential(
    (0): Flatten(start_dim=1, end_dim=-1)
    (1): Linear(in_features=784, out_features=64, bias=True)
    (2): Sigmoid()
  )
  (layer2): Sequential(
    (0): Linear(in_features=64, out_features=32, bias=True)
    (1): Sigmoid()
  )
  (layer3): Sequential(
    (0): Linear(in_features=32, out_features=16, bias=True)
    (1): Sigmoid()
  )
  (layer4): Sequential(
    (0): Linear(in_features=16, out_features=10, bias=True)
  )
  (fc): Linear(in_features=1568, out_features=10, bias=True)
)
Traceback (most recent call last):
  File "C:\Users\bjark\AppData\Local\Temp\ipykernel_25668\30544962.py", line 13, in model_pipeline
    train(model, train_loader, criterion, optimizer, config)
  File "C:\Users\bjark\AppData\Local\Temp\ipykernel_25668\2177316483.py", line 12, in train
    loss = train_batch(images, labels, model, optimizer, criterion)
  File "C:\Users\bjark\AppData\Local\Temp\ipykernel_25668\2177316483.py", line 25, in train_batch
    outputs = model(images)
  File "C:\Users\bjark\anaconda3\lib\site-packages\torch\nn\modules\module.py", line 1538, in _call_impl
    result = forward_call(*args, **kwargs)
  File "C:\Users\bjark\AppData\Local\Temp\ipykernel_25668\1031426566.py", line 29, in forward
    out = self.fc(out)
  File "C:\Users\bjark\anaconda3\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "C:\Users\bjark\anaconda3\lib\site-packages\torch\nn\modules\linear.py", line 114, in forward
    return F.linear(input, self.weight, self.bias)
RuntimeError: mat1 and mat2 shapes cannot be multiplied (128x10 and 1568x10)